docs/troubleshooting.md

```diff
@@ -0,0 +1,471 @@
+     --assets tokenizer \
+     --hf_token=YOUR_TOKEN_HERE
+   ```
+
+### Import Errors
+
+**Problem:** Module not found errors
+```
+ModuleNotFoundError: No module named 'torchtitan'
+```
+
+**Solutions:**
+1. Install in development mode:
+   ```bash
+   pip install -e .
+   ```
+
+2. Or add to PYTHONPATH:
+   ```bash
+   export PYTHONPATH="${PYTHONPATH}:$(pwd)"
+   ```
+
+### Pre-commit Hooks Fail
+
+**Problem:** Pre-commit checks fail on commit
+
+**Solutions:**
+1. Install pre-commit hooks:
+   ```bash
+   pip install pre-commit
+   pre-commit install
+   ```
+
+2. Fix issues automatically:
+   ```bash
+   pre-commit run --all-files
+   ```
+
+3. Update hooks if outdated:
+   ```bash
+   pre-commit autoupdate
+   ```
+
+---
+
+## Training Issues
+
+### Out of Memory (OOM) Errors
+
+**Problem:** CUDA out of memory during training
+```
+RuntimeError: CUDA out of memory
+```
+
+**Solutions:**
+
+1. **Reduce batch size** in your config:
+   ```toml
+   [training]
+   batch_size = 1  # Try smaller batch size
+   ```
+
+2. **Enable activation checkpointing**:
+   ```toml
+   [activation_checkpoint]
+   mode = 'selective'  # or 'full'
+   ```
+
+3. **Use gradient accumulation** instead of larger batches:
+   ```toml
+   [training]
+   batch_size = 1
+   global_batch_size = 8  # Effective batch size through accumulation
+   ```
+
+4. **Enable FSDP2 with CPU offload**:
+   ```toml
+   [model]
+   fsdp_cpu_offload = true
+   ```
+
+5. **Profile memory** to identify bottleneck:
+   ```bash
+   CONFIG_FILE="./train_configs/debug_model.toml" ./run_train.sh \
+     --profiling.enable_memory_snapshot \
+     --profiling.save_memory_snapshot_folder memory_snapshot
+   ```
+   Then visualize at https://pytorch.org/memory_viz
+
+### Training Hangs at Start
+
+**Problem:** Training process hangs without error messages
+
+**Solutions:**
+
+1. **Enable debug logging**:
+   ```bash
+   export TORCH_DISTRIBUTED_DEBUG=DETAIL
+   export NCCL_DEBUG=INFO
+   ```
+
+2. **Check GPU availability** on all nodes:
+   ```bash
+   nvidia-smi
+   ```
+
+3. **Verify network connectivity** (multi-node):
+   ```bash
+   # Test from node 0 to node 1
+   ping <node1-ip>
+   ```
+
+4. **Start with debug model**:
+   ```bash
+   CONFIG_FILE="./torchtitan/models/llama3/train_configs/debug_model.toml" ./run_train.sh
+   ```
+
+### Checkpoint Loading Fails
+
+**Problem:** Cannot load checkpoint
+```
+FileNotFoundError: Checkpoint not found
+```
+
+**Solutions:**
+
+1. **Verify checkpoint path** in config:
+   ```toml
+   [checkpoint]
+   folder = "./outputs/checkpoint"
+   ```
+
+2. **Check checkpoint compatibility**:
+   - Ensure same model architecture
+   - Verify parallelism settings match
+
+3. **Use checkpoint conversion** if from HuggingFace:
+   ```bash
+   python scripts/checkpoint_conversion/convert_from_hf.py \
+     --model_name llama3 \
+     --hf_checkpoint_path /path/to/hf \
+     --output_path ./checkpoint
+   ```
+
+### Loss is NaN or Diverging
```

Comment:
Also this issue might be very different under different cases: Simpler case might be learning rate is too high,  but sometimes it even because of a pytorch or torchtitan bug